### PR TITLE
[SYCLomatic] Fix pulldown conflicts  and lit test failure from sycl branch to SYCLomatic branch

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/include/clang/Driver/Options.td [hunk 1]
```diff
@@ -2028,15 +2028,11 @@ defm emulated_tls : BoolFOption<"emulated-tls",
...
 // SYCLomatic_CUSTOMIZATION begin
-def ferror_limit_EQ : Joined<["-"], "ferror-limit=">, Group<f_Group>, Flags<[CoreOption]>,
+def ferror_limit_EQ : Joined<["-"], "ferror-limit=">, Group<f_Group>, Visibility<[ClangOption, CLOption, DXCOption]>,
   HelpText<"Set the maximum number of errors to emit before stopping (0 = no limit).">;
 // SYCLomatic_CUSTOMIZATION end
-=======
-def ferror_limit_EQ : Joined<["-"], "ferror-limit=">, Group<f_Group>,
-  Visibility<[ClangOption, CLOption, DXCOption]>;
->>>>>>> origin/sycl
+
```
Comment (zhiweij1) [inline clang/include/clang/Driver/Options.td]: Remove extra line

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 2]
```diff
@@ -733,10 +733,14 @@ void DpctFileInfo::buildLinesInfo() {
     return;
   auto &SM = DpctGlobalInfo::getSourceManager();
 
-  auto FE = SM.getFileManager().getFile(FilePath);
-  if (std::error_code ec = FE.getError())
+  llvm::Expected<FileEntryRef> Result =
+      SM.getFileManager().getFileRef(FilePath);
+
+  if (auto E = Result.takeError()) {
```
Comment (zhiweij1) [inline clang/lib/DPCT/AnalysisInfo.cpp]: if (auto E = Result.takeError())

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 3]
```diff
@@ -733,10 +733,14 @@ void DpctFileInfo::buildLinesInfo() {
...
     return;
-  auto FID = SM.getOrCreateFileID(FE.get(), SrcMgr::C_User);
+  }
```
Comment (zhiweij1) [inline clang/lib/DPCT/AnalysisInfo.cpp]: Remove

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 4]
```diff
@@ -733,10 +733,14 @@ void DpctFileInfo::buildLinesInfo() {
...
+
```
Comment (zhiweij1) [inline clang/lib/DPCT/AnalysisInfo.cpp:741]: Remove extra line

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 5]
```diff
@@ -733,10 +733,14 @@ void DpctFileInfo::buildLinesInfo() {
...
+  auto FID = SM.getOrCreateFileID(Result.get(), SrcMgr::C_User);
```
Comment (zhiweij1) [inline clang/lib/DPCT/AnalysisInfo.cpp]: `auto FID = SM.getOrCreateFileID(*Result, SrcMgr::C_User);` ?

File: clang/lib/DPCT/Diagnostics.h [hunk 6]
```diff
@@ -402,8 +402,16 @@ bool report(const std::string &FileAbsPath, unsigned int Offset, IDTy MsgID,
     return false;
 
   SourceManager &SM = SourceManagerForWarning::getSM();
-  FileID FID = SM.getOrCreateFileID(
-      SM.getFileManager().getFile(NativeFormPath).get(), SrcMgr::C_User);
+
+  llvm::Expected<FileEntryRef> Result =
```
Comment (zhiweij1) [inline clang/lib/DPCT/Diagnostics.h:406]:
```
  llvm::Expected<FileEntryRef> Result =
      SM.getFileManager().getFileRef(NativeFormPath);
  if (auto E = Result.takeError())
    return false;
  FileID FID = SM.getOrCreateFileID(*Result, SrcMgr::C_User);
```

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 7]
```diff
@@ -553,19 +553,24 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool, StringRef InRoot,
...
+      if (auto E = Result.takeError()) {
+        continue;
+      }
+
       if (MapNames::PatternRewriters.empty()) {
         Rewrite
             .getEditBuffer(Sources.getOrCreateFileID(
-                Tool.getFiles().getFile(Entry.first).get(),
-                clang::SrcMgr::C_User /*normal user code*/))
+                Result.get(), clang::SrcMgr::C_User /*normal user code*/))
```
Comment (zhiweij1) [inline clang/lib/DPCT/SaveNewFiles.cpp]: Result.get() => *Result

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 8]
```diff
@@ -553,19 +553,24 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool, StringRef InRoot,
...
             .write(Stream);
       } else {
         std::string OutputString;
         llvm::raw_string_ostream RSW(OutputString);
         Rewrite
             .getEditBuffer(Sources.getOrCreateFileID(
-                Tool.getFiles().getFile(Entry.first).get(),
-                clang::SrcMgr::C_User /*normal user code*/))
+                Result.get(), clang::SrcMgr::C_User /*normal user code*/))
```
Comment (zhiweij1) [inline clang/lib/DPCT/SaveNewFiles.cpp]: Result.get() => *Result

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 9]
```diff
@@ -711,19 +716,22 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool, StringRef InRoot,
...
+          Tool.getFiles().getFileRef(Entry.first);
+      if (auto E = Result.takeError()) {
+        continue;
+      }
       if (MapNames::PatternRewriters.empty()) {
         Rewrite
             .getEditBuffer(Sources.getOrCreateFileID(
-                Tool.getFiles().getFile(Entry.first).get(),
-                clang::SrcMgr::C_User /*normal user code*/))
+                Result.get(), clang::SrcMgr::C_User /*normal user code*/))
```
Comment (zhiweij1) [inline clang/lib/DPCT/SaveNewFiles.cpp]: Result.get() => *Result

File: clang/lib/DPCT/SaveNewFiles.cpp [hunk 10]
```diff
@@ -711,19 +716,22 @@ int saveNewFiles(clang::tooling::RefactoringTool &Tool, StringRef InRoot,
...
             .write(Stream);
       } else {
         std::string OutputString;
         llvm::raw_string_ostream RSW(OutputString);
         Rewrite
             .getEditBuffer(Sources.getOrCreateFileID(
-                Tool.getFiles().getFile(Entry.first).get(),
-                clang::SrcMgr::C_User /*normal user code*/))
+                Result.get(), clang::SrcMgr::C_User /*normal user code*/))
```
Comment (zhiweij1) [inline clang/lib/DPCT/SaveNewFiles.cpp]: Result.get() => *Result

File: clang/lib/Tooling/Core/Replacement.cpp [hunk 11]
```diff
@@ -122,17 +122,17 @@ void Replacement::setFromSourceLocation(const SourceManager &Sources,
                                         StringRef ReplacementText) {
   const std::pair<FileID, unsigned> DecomposedLocation =
       Sources.getDecomposedLoc(Start);
-<<<<<<< HEAD
-  const FileEntry *Entry = Sources.getFileEntryForID(DecomposedLocation.first);
```
Comment (zhiweij1) [inline clang/lib/Tooling/Core/Replacement.cpp:125]: EntryRef = Sources.getFileEntryRefForID(DecomposedLocation.first);

File: clang/lib/Tooling/Core/Replacement.cpp [hunk 12]
```diff
@@ -122,17 +122,17 @@ void Replacement::setFromSourceLocation(const SourceManager &Sources,
...
 #ifdef SYCLomatic_CUSTOMIZATION
-  if (Entry) {
+  if (Sources.getFileEntryRefForID(DecomposedLocation.first).has_value()) {
```
Comment (zhiweij1) [inline clang/lib/Tooling/Core/Replacement.cpp]: if (EntryRef.has_value()) {

File: clang/lib/Tooling/Core/Replacement.cpp [hunk 13]
```diff
@@ -122,17 +122,17 @@ void Replacement::setFromSourceLocation(const SourceManager &Sources,
...
+  const auto &Entry =
```
Comment (zhiweij1) [inline clang/lib/Tooling/Core/Replacement.cpp]: const auto &Entry = EntryRef->getFileEntry();

File: clang/lib/DPCT/AnalysisInfo.cpp (hunk 5 above)
Comment (tomflinda) [reply clang/lib/DPCT/AnalysisInfo.cpp]: OKay to me

File: clang/lib/DPCT/Diagnostics.h (hunk 6 above)
Comment (tomflinda) [reply clang/lib/DPCT/Diagnostics.h:406]: okay

File: clang/lib/DPCT/SaveNewFiles.cpp (hunk 8 above)
Comment (tomflinda) [reply clang/lib/DPCT/SaveNewFiles.cpp]: OKay

File: clang/lib/DPCT/SaveNewFiles.cpp (hunk 9 above)
Comment (tomflinda) [reply clang/lib/DPCT/SaveNewFiles.cpp]: ok

File: clang/lib/DPCT/SaveNewFiles.cpp (hunk 10 above)
Comment (tomflinda) [reply clang/lib/DPCT/SaveNewFiles.cpp]: okay

**Review by zhiweij1 — APPROVED** [review]

File: llvm/utils/git/requirements.txt.in [hunk 14]
```diff
@@ -3,11 +3,5 @@
 # pip install pip-tools
 # pip-compile -o requirements.txt  requirements.txt.in
 
-<<<<<<< HEAD
 PyGithub
 #GitPython==3.1.29
-=======
```
Comment (zhimingwang36) [inline llvm/utils/git/requirements.txt.in]: please use the set from Pulldown which has latest setting.

Comment (tomflinda) [reply llvm/utils/git/requirements.txt.in]: After a further look, line8 has been commented in SYCLomatic, we only keep line7.

Comment (zhimingwang36) [reply llvm/utils/git/requirements.txt.in]: LGTM

**Pull request merged**